src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/mappers/TripPatternForDateMapper.java

```diff
@@ -53,6 +53,18 @@ public class TripPatternForDateMapper {
   public TripPatternForDate map(Timetable timetable, LocalDate serviceDate) {
     TIntSet serviceCodesRunning = serviceCodesRunningForDate.get(serviceDate);
 
+    // ServiceCodesRunning is potentially null if the date is not present in the original GTFS.
+    // At that point we can simply return null, as there are no trips running on that date.
+    if (serviceCodesRunning == null) {
```

Comment:
This looks ok, but can you please add a test for this code?

Reply:
If you're not sure how to do it, get in touch and I can help.

Reply:
I've added a test for my new addition, and also for the whole class as a whole, as it wasn't tested at all before.